storage/src/adb/keyless.rs

```diff
@@ -0,0 +1,863 @@
+    /// The name of the [Storage] partition used for the MMR's metadata.
+    pub mmr_metadata_partition: String,
+
+    /// The name of the [Storage] partition used to persist the (pruned) values.
+    pub values_journal_partition: String,
+
+    /// The size of the write buffer to use with the values journal.
+    pub values_write_buffer: NonZeroUsize,
+
+    /// Optional compression level (using `zstd`) to apply to values data before storing.
+    pub values_compression: Option<u8>,
+
+    /// The codec configuration to use for encoding and decoding values.
+    pub values_codec_config: C,
+
+    /// The max number of values to put in each section of the values journal.
+    pub values_items_per_section: NonZeroU64,
+
+    /// The name of the [Storage] partition used for the location map.
+    pub locations_journal_partition: String,
+
+    /// The number of items to put in each blob in the location map.
+    pub locations_items_per_blob: NonZeroU64,
+
+    /// The size of the write buffer to use with the locations journal.
+    pub locations_write_buffer: NonZeroUsize,
+
+    /// An optional thread pool to use for parallelizing batch operations.
+    pub thread_pool: Option<ThreadPool>,
+
+    /// The buffer pool to use for caching data.
+    pub buffer_pool: PoolRef,
+}
+
+/// A keyless ADB for variable length data.
+pub struct Keyless<E: Storage + Clock + Metrics, V: Codec, H: CHasher> {
+    /// An MMR over digests of the operations applied to the db.
+    ///
+    /// # Invariant
+    ///
+    /// The number of leaves in this MMR always equals the number of operations in the unpruned
+    /// `locations` journal.
+    mmr: Mmr<E, H>,
+
+    /// A (pruned) journal of all values ever appended to the db.
+    values: VJournal<E, V>,
+
+    /// The total number of values appended (including those that have been pruned).  The next
+    /// appended item will have this value as its location.
+    size: u64,
+
+    /// The number of values to put in each section of the values journal.
+    ///
+    /// Unlike the other variable-type stores, the actual number of values could be less than this
+    /// amount even if the section is "full", since we don't explicitly insert anything into the
+    /// section for commits.
+    values_per_section: u64,
+
+    /// A fixed-length journal that maps an appended value's location to its offset within its
+    /// respective section of the values journal. (The section number is derived from location.)
+    ///
+    /// The locations structure provides the "source of truth" for the db's pruning boundaries and
+    /// overall size, should there be any discrepancies.
+    locations: FJournal<E, Operation>,
+
+    /// Cryptographic hasher to re-use within mutable operations requiring digest computation.
+    hasher: Standard<H>,
+}
+
+impl<E: Storage + Clock + Metrics, V: Codec, H: CHasher> Keyless<E, V, H> {
+    /// Returns a [Keyless] adb initialized from `cfg`. Any uncommitted operations will be discarded
```

Comment:
nit: `On restart, any uncommitted...`

Reply:
This is documenting what this function actually does, not what happens on restart, so I think the comment is more correct as-is?